### PR TITLE
New module jvarkit/wgscoverageplotter

### PR DESCRIPTION
**Pull request by lindenb**
This PR for  jvarkit/wgscoverageplotter  https://jvarkit.readthedocs.io/en/latest/WGSCoveragePlotter/  that was cited in

```
    Vinicius A.C & al. Comparative analyses of Theobroma cacao and T. grandiflorum mitogenomes reveal conserved gene content embedded within complex and plastic structures". Gene Volume 849 2023 .https://doi.org/10.1016/j.gene.2022.146904.
    Varani AM, Silva SR, Lopes S, Barbosa JBF, Oliveira D, Correa MA, Moraes AP, Miranda VFO, Prosdocimi F. 2022. The complete organellar genomes of the entheogenic plant Psychotria viridis (Rubiaceae), a main component of the ayahuasca brew. PeerJ 10:e14114 https://doi.org/10.7717/peerj.14114
    Cho, C.H., Park, S.I., Huang, TY. et al. Genome-wide signatures of adaptation to extreme environments in red algae. Nat Commun 14, 10 (2023). https://doi.org/10.1038/s41467-022-35566-x
    Rhys T. White, Martina Jelocnik, Natalie Klukowski, Md. Hakimul Haque, Subir Sarker. The first genomic insight into Chlamydia psittaci sequence type (ST)24 from a healthy captive psittacine host in Australia demonstrates evolutionary proximity with strains from psittacine, human, and equine hosts. Veterinary Microbiology, Volume 280, 2023, 109704, ISSN 0378-1135. https://doi.org/10.1016/j.vetmic.2023.109704.
    Rhys T. White. A discovery down under: decoding the draft genome sequence of Pantoea stewartii from Australia's Critically Endangered western ground parrot/kyloring (Pezoporus flaviventris). https://doi.org/10.1099/mgen.0.001101 . Microbial Genomics Vol9 Issue 9
    genomic and transcriptomic insight into the evolution and breeding of turfgrass species, poa annua and bouteloua dactyloides. Thesis. Christopher W. Benson. https://etda.libraries.psu.edu/files/final_submissions/29057 WGSCoveragePlotter
    Jo, J., Park, JS., Won, H. et al. The first Chromosomal-level genome assembly of Sageretia thea using Nanopore long reads and Pore-C technology. Sci Data 11, 959 (2024). https://doi.org/10.1038/s41597-024-03798-9
```

## PR checklist

Closes #XXX 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/jvarkit/wgscoverageplotter/meta.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        e.g. [ id:'test_reference' ]
+  - fasta:
+      type: file
+      description: The reference fasta file
+      pattern: "*.fasta"
+  - meta3:
+      type: map
+      description: |
+        Groovy Map containing reference fai information
+      pattern: "*.fai"
```
Comment (maxulysse) [inline modules/nf-core/jvarkit/wgscoverageplotter/meta.yml:48]:
```suggestion
      e.g. [ id:'test_reference' ]
```

File: modules/nf-core/jvarkit/wgscoverageplotter/meta.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  - fai:
+      type: file
+      description: Index of reference fasta file
+      pattern: "fasta.fai"
+  - meta4:
+      type: map
+      description: |
+        Groovy Map containing reference dict information
+      pattern: "*.dict"
```
Comment (maxulysse) [inline modules/nf-core/jvarkit/wgscoverageplotter/meta.yml:57]:
```suggestion
      e.g. [ id:'test_reference' ]
```

File: modules/nf-core/jvarkit/wgscoverageplotter/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                            path(process.out.output[0][1]),
+                            process.out.versions
+                            ).match()
+                }
```
Comment (maxulysse) [inline modules/nf-core/jvarkit/wgscoverageplotter/tests/main.nf.test:39]:
```suggestion
                { assert snapshot(process.out).match() }
```

File: modules/nf-core/jvarkit/wgscoverageplotter/tests/main.nf.test [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(
+                       path(process.out.output[0][1]),
+                       process.out.versions
+                       ).match()
+                }
```
Comment (maxulysse) [inline modules/nf-core/jvarkit/wgscoverageplotter/tests/main.nf.test:73]:
```suggestion
                { assert snapshot(process.out).match() }
```

**Review by maxulysse — COMMENTED** [review]
minor comments

**Comment by lindenb** [comment]
oh the full path of the input file is injected in the final XML+SVG file. So the MD5 might change... I asked for help here: https://nfcore.slack.com/archives/CJRH30T6V/p1726166942126159

File: modules/nf-core/jvarkit/wgscoverageplotter/main.nf [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    tuple val(meta4), path(dict)
+    output:
+    tuple val(meta),  path("*.svg"), emit: output
+    path "versions.yml"            , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args1        = task.ext.args1 ?: ''
```
Comment (SPPearce) [inline modules/nf-core/jvarkit/wgscoverageplotter/main.nf:23]:
```suggestion
    def args        = task.ext.args ?: ''
```

File: modules/nf-core/jvarkit/wgscoverageplotter/main.nf [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    def prefix       = task.ext.prefix ?: "${meta.id}"
+    """
+    mkdir -p TMP
+
+    jvarkit -Xmx${task.memory.giga}g  -XX:-UsePerfData -Djava.io.tmpdir=TMP wgscoverageplotter \\
+        -R ${fasta} \\
+        ${args1} \\
```
Comment (SPPearce) [inline modules/nf-core/jvarkit/wgscoverageplotter/main.nf:30]:
```suggestion
        ${args} \\
```

File: modules/nf-core/jvarkit/wgscoverageplotter/main.nf [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    rm -rf TMP
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        jvarkit: \$(jvarkit -v)
+    END_VERSIONS
+    """
+
+    stub:
+    def args2        = task.ext.args2 ?: ''
```
Comment (SPPearce) [inline modules/nf-core/jvarkit/wgscoverageplotter/main.nf:42]:
```suggestion
```
Don't need args at all for the stub.

File: modules/nf-core/jvarkit/wgscoverageplotter/tests/main.nf.test [hunk 8]
```diff
@@ -0,0 +1,71 @@
+// nf-core modules test jvarkit/wgscoverageplotter
```
Comment (SPPearce) [inline modules/nf-core/jvarkit/wgscoverageplotter/tests/main.nf.test:1]:
```suggestion
```

**Review by SPPearce — APPROVED** [review]
Looks good to me.

**Pull request merged**